azure-iot-device/tests/iothub/pipeline/test_iothub_pipeline.py

```diff
@@ -145,7 +145,7 @@ def test_sas_auth_op_fail(self, mocker, device_connection_string, fake_exception
         old_execute_op = pipeline_stages_base.PipelineRootStage._execute_op
 
         def fail_set_auth_provider(self, op):
-            if isinstance(op, pipeline_stages_base.SetAuthProviderOperation):
```

Comment:
How did this test ever pass? same question (and probably same answer) for the change on line 185 below.

Reply:
That is a very, very good question. I was shocked to find this.

---
In reply to: [326325693](https://github.com/Azure/azure-iot-sdk-python/pull/309#discussion_r326325693) [](ancestors = 326325693)